lib/sqlalchemy/sql/schema.py

```diff
@@ -1307,6 +1315,11 @@ def referred_schema_fn(table, to_schema,
                 _table=table,
                 **index.kwargs,
             )
+        for period in self.periods:
```

Comment:
I see you have the start of a `period._copy()' method below, so I assume the period-specific logic part of this will be moved there.